### PR TITLE
C++: Add a taint model for `realloc`

### PR DESCRIPTION
**Pull request by MathiasVP**
This should fix some missing results that @jketema was observing on MRVA 🤞

### CODE REVIEW

**Review by geoffw0 — APPROVED** [review]
LGTM!

**Comment by geoffw0** [comment]
(we should probably wait for the MRVA run to finish before merging)

**Comment by MathiasVP** [comment]
> (we should probably wait for the MRVA run to finish before merging)

Indeed, I'll check up on DCA tomorrow morning before I merge this. I have a feeling this will create some new results on taint-tracking queries.

**Comment by geoffw0** [comment]
Ping me if you'd like an opinion / second opinion on any affected results.

**Comment by MathiasVP** [comment]
The new flow exposed some FPs on the `cpp/potential-system-data-exposure` query on SAMATE caused by the lack of any barriers in the query. I've added the minimum set of barriers that makes SAMATE happy in 4a1bf95a877c0cbf9ca93a6fbbc881c4e0b08e43.

File: cpp/ql/lib/semmle/code/cpp/models/implementations/Memset.qll [hunk 1]
```diff
@@ -9,18 +9,17 @@ import semmle.code.cpp.models.interfaces.DataFlow
...
 {
-  MemsetFunction() {
+  MemsetFunctionModel() {
     this.hasGlobalOrStdOrBslName("memset")
     or
     this.hasGlobalOrStdName("wmemset")
     or
-    this.hasGlobalName([bzero(), "__builtin_memset", "__builtin_memset_chk"])
+    this.hasGlobalName([
+        bzero(), "__builtin_memset", "__builtin_memset_chk", "RtlZeroMemory", "RtlSecureZeroMemory"
```
Comment (geoffw0) [inline cpp/ql/lib/semmle/code/cpp/models/implementations/Memset.qll:21]: Notably on some build configurations `RtlSecureZeroMemory` is a macro, but it's a macro for `memset` and that is modelled already.

**Review by geoffw0 — APPROVED** [review]
Changes LGTM. :+1:

File: cpp/ql/lib/semmle/code/cpp/models/implementations/Memset.qll (hunk 1 above)
Comment (MathiasVP) [reply cpp/ql/lib/semmle/code/cpp/models/implementations/Memset.qll:21]: Correct. Not in Samate, though 😂!

**Comment by MathiasVP** [comment]
The two lost results changes on `vim` is due to the default value for `fieldFlowBranchLimit`. Bumping the limit to `3` gives us back the two results. I'm not worried about this since this is losing 2 out of ~1500 results on that query on `vim`. So I think this is good to go 🎉

**Pull request merged**